assets/js/langRedirector.js

```diff
@@ -0,0 +1,49 @@
+// Wrap all this to not pollute global namespace
+{
+    // Generate an array of languages from page
+    const JDM_LANGS = [...document.querySelectorAll('#language-dropdown > ul > li')].map(el => el.classList[0]);
+
+
+    function redirectTo(lang) {
+        const link = document.querySelector(`#language-dropdown > ul > li.${lang} > a`);
+        if (link)
+            link.click();
+    }
+
+
+    function guessedTranslation() {
+        const langs = navigator.languages;
+
+        for (let i=0; i < langs.length; i++) {
+            let lang = langs[i];
+
+            // user has: 'pt-BR'
+            // we have:  'pt-BR'
+            if (JDM_LANGS.includes(lang))
+                return lang;
+
+            // user has: 'ru-RU'
+            // we have:  'ru'
+            lang = lang.slice(0,2);
+            if (JDM_LANGS.includes(lang))
+                return lang;
+
+            // user has: 'pt-AO' or 'pt'
+            // we have:  'pt-BR'
+            for (let j=0; j < JDM_LANGS.length; j++) {
+                let jdm_lang = JDM_LANGS[j];
+                if (jdm_lang.slice(0,2) === lang)
+                    return jdm_lang;
+            }
+        }
+
+        return null;
+    }
```

Comment:
```suggestion
    function exactLangMatch(lang) {
        // user has: 'pt-BR'
        // we have:  'pt-BR'
        return (JDM_LANGS.includes(lang) ? lang : null);
    }
    
    function partialLangMatch(lang) {
        // user has: 'ru-RU'
        // we have:  'ru'
        lang = lang.slice(0,2);
        return exactLangMatch(lang);
    }
    
    function roughLangMatch(lang) {
        // user has: 'pt-AO' or 'pt'
        // we have:  'pt-BR'
        for (let j=0; j < JDM_LANGS.length; j++) {
            let jdm_lang = JDM_LANGS[j];
            if (jdm_lang.slice(0,2) === lang)
                return jdm_lang;
        }
        return null;
    }

    function guessedTranslation() {
        const langs = navigator.languages;

        for (let i=0; i < langs.length; i++) {
            let lang = langs[i];

            let exactLang = exactLangMatch(lang);
            if (exactLang !== null)
                return exactLang;
            
            let partialLang = partialLangMatch(lang);
            if (partialLang !== null)
                return partialLang;

            let roughLang = roughLangMatch(lang.slice(0,2));
            if (roughLang !== null)
                return roughLang;
        }

        return null;
    }
```

I'd suggest extracting each of the matches into a function. For the second one we can actually reuse the first and it moves the comments there. I don't particularly like all the `if !== null`, but I think overall it's clearer. Let me know what you guys think.